### PR TITLE
Added announcements endpoints with swagger doc completed for it

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
image

Added the announcements CRUD endpoints with the swagger documentation for it 

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

Terminal commands to test code:-
1). npm run lint
2). npm run lint:fix
3). nest start
4). npm run test

### CODE REVIEW

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09  The Announcements CRUD with the schema, dtos and its swagger documentation for the APIs

File: src/modules/announcements/announcement.controller.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  @ApiProperty({ required: true, example: 'Noob' })
+  name: string;
+
+  @ApiProperty({ required: true, example: 'Coder' })
+  link: string;
+
+  @ApiProperty({ required: true, example: 'noobcoder@gmai.com' })
+  submit_by: string;
+}
```
Comment (Abhishek-kumar09) [inline src/modules/announcements/announcement.controller.ts:35]: Remove this

File: src/modules/announcements/announcement.controller.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+@ApiTags('Announcement')
+@Controller('Announcement')
+export class AnnouncementController {
+  constructor(private AnnouncementService: AnnouncementService) {}
+
+  // add a Announcement
+  @Post()
+  @ApiCreatedResponse(responsedoc.addAnnouncement)
+  async addAnnouncement(@Body() CreateAnnouncementDTO: CreateAnnouncementDTO) {
+    return await this.AnnouncementService.addAnnouncement(
```
Comment (Abhishek-kumar09) [inline src/modules/announcements/announcement.controller.ts:46]: camelcasing in service name

File: src/modules/announcements/announcement.controller.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  @ApiOkResponse(responsedoc.getAllAnnouncement)
+  async getAllAnnouncement() {
+    return await this.AnnouncementService.getAllAnnouncement();
+  }
+
+  // Fetch a particular Announcement using ID
+  @Get('/:AnnouncementId')
+  @ApiParam(AnnouncementId)
+  @ApiOkResponse({ type: AnnouncementResponseBody })
+  async getAnnouncement(@Param('AnnouncementId') AnnouncementId: string) {
```
Comment (Abhishek-kumar09) [inline src/modules/announcements/announcement.controller.ts:62]: Please use camel casing in variable naming.

File: src/modules/announcements/announcement.service.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      throw new InternalServerErrorException(e);
+    }
+  }
+
+  // Get a single Announcement
+  async getAnnouncement(AnnouncementId): Promise<Announcement> {
+    try {
+      const Announcement = await this.AnnouncementModel.findById(
+        AnnouncementId,
+      ).exec();
```
Comment (Abhishek-kumar09) [inline src/modules/announcements/announcement.service.ts:34]: remove `.exec()`

File: src/modules/announcements/announcement.controller.ts (hunk 1 above)
Comment (Devesh21700Kumar) [reply src/modules/announcements/announcement.controller.ts:35]: ok

File: src/modules/announcements/announcement.controller.ts (hunk 2 above)
Comment (Devesh21700Kumar) [reply src/modules/announcements/announcement.controller.ts:46]: ok

File: src/modules/announcements/announcement.controller.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/modules/announcements/announcement.controller.ts:62]: ok

File: src/modules/announcements/announcement.service.ts (hunk 4 above)
Comment (Devesh21700Kumar) [reply src/modules/announcements/announcement.service.ts:34]: ok

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09  changes have been made

File: src/modules/announcements/dto/create-announcement.dto.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { IsBoolean, IsNotEmpty, IsString } from 'class-validator';
+
+export class CreateAnnouncementDTO {
```
Comment (Abhishek-kumar09) [inline src/modules/announcements/dto/create-announcement.dto.ts:3]: Add creator:
`created_by`

Comment (Devesh21700Kumar) [reply src/modules/announcements/dto/create-announcement.dto.ts:3]: Added it

**Pull request merged**